src/main/kotlin/com/sourcegraph/cody/chat/ContextFilesMessage.kt

```diff
@@ -0,0 +1,73 @@
+package com.sourcegraph.cody.chat
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.project.DumbAwareAction
+import com.intellij.openapi.project.Project
+import com.intellij.ui.components.AnActionLink
+import com.intellij.util.ui.JBInsets
+import com.intellij.util.ui.JBUI
+import com.sourcegraph.cody.agent.protocol.ContextFile
+import com.sourcegraph.cody.agent.protocol.ContextMessage
+import com.sourcegraph.cody.agent.protocol.Speaker
+import com.sourcegraph.cody.chat.ChatUIConstants.*
+import com.sourcegraph.cody.ui.AccordionSection
+import java.awt.BorderLayout
+import java.awt.Insets
+import java.util.*
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.stream.Collectors
+import javax.swing.JPanel
+import javax.swing.border.EmptyBorder
+
+class ContextFilesMessage(project: Project, contextMessages: List<ContextMessage>) :
+    PanelWithGradientBorder(ASSISTANT_MESSAGE_GRADIENT_WIDTH, Speaker.ASSISTANT) {
+  init {
+    this.layout = BorderLayout()
+
+    val margin = JBInsets.create(Insets(TEXT_MARGIN, TEXT_MARGIN, TEXT_MARGIN, TEXT_MARGIN))
+    val contextFileNames =
+        contextMessages
+            .stream()
+            .map<ContextFile>(ContextMessage::file)
+            .filter { obj: ContextFile? -> Objects.nonNull(obj) }
+            .map(ContextFile::fileName)
+            .collect(Collectors.toSet())
+
+    val accordionSection = AccordionSection("Read ${contextFileNames.size} files")
+    accordionSection.isOpaque = false
+    accordionSection.border = EmptyBorder(margin)
+    val fileIndex = AtomicInteger(0)
+    contextFileNames.forEach { fileName: String ->
+      val filePanel = createFileWithLinkPanel(project, fileName)
+      accordionSection.contentPanel.add(filePanel, fileIndex.getAndIncrement())
+    }
+    add(accordionSection, BorderLayout.CENTER)
+  }
+
+  private fun createFileWithLinkPanel(project: Project, fileName: String): JPanel {
+    val fileWithoutRedundantPrefix =
+        fileName.removePrefix("../../../..").removePrefix(project.baseDir.path)
```

Comment:
Can you pls elaborate on this `removePrefix("../../../..")`?  It looks like a underlying problem. Are we getting such paths from the agent?

Reply:
nvm, I was using a wrong repo as a context

Reply:
or... there is a problem 🤔

Reply:
![image](https://github.com/sourcegraph/jetbrains/assets/19799111/904f6ada-d4b0-48ae-a468-21ddac57be01)

installed on my IDE & run in jetbrains project

Reply:
nah, I probably had the former bad version or bad context repo - it works properly now.

there is another problem, however - I will start a discussion below